src/github.com/getlantern/peerscanner/web.go

```diff
@@ -183,3 +218,15 @@ var fallbackIPPrefixes = []string{
 	"178.62",
 	"188.166",
 }
+
+func getSingleFormValue(req *http.Request, name string) string {
+	ls := req.Form[name]
+	if len(ls) == 0 {
+		return ""
+	}
+	if len(ls) > 1 {
+		// But we still allow it for robustness.
+		log.Errorf("More than one '%v' provided in form: %v", name, ls)
+	}
+	return ls[0]
+}
```

Comment:
[gorilla](http://www.gorillatoolkit.org/) or something similar may already provide a lot of this kind of functionality.